Re-enable no-error-pmd build

We dropped the no-error-pmd build because there were failures due to https://github.com/checkstyle/checkstyle/pull/15376.
We should re enable this build when PMD merge the PR that uses the new property to fix the violations. The PR can't be merged until checkstyle new release.
PMD PR: https://github.com/pmd/build-tools/pull/33

build failure URL: https://app.circleci.com/pipelines/github/checkstyle/checkstyle/27241/workflows/ae4881cd-8f64-4245-a7c0-619e023f8e5e/jobs/662368

```
There are 8 errors reported by Checkstyle 10.18.0-SNAPSHOT with /net/sourceforge/pmd/pmd-checkstyle-config.xml ruleset.
[ERROR] src/main/java/net/sourceforge/pmd/lang/java/ast/ConstantFolder.java:[24,7] (modifier) RedundantModifier: Redundant 'strictfp' modifier.
[ERROR] src/main/java/net/sourceforge/pmd/lang/java/ast/ConstantFolder.java:[141,12] (modifier) RedundantModifier: Redundant 'strictfp' modifier.
...
```

```
[INFO] There are 8 errors reported by Checkstyle 10.18.0-SNAPSHOT with /net/sourceforge/pmd/pmd-checkstyle-config.xml ruleset.
[ERROR] src/main/java/net/sourceforge/pmd/lang/java/ast/ConstantFolder.java:[24,7] (modifier) RedundantModifier: Redundant 'strictfp' modifier.
[ERROR] src/main/java/net/sourceforge/pmd/lang/java/ast/ConstantFolder.java:[141,12] (modifier) RedundantModifier: Redundant 'strictfp' modifier.
[ERROR] src/test/java/net/sourceforge/pmd/lang/java/ast/ConstantExpressionsTests.java:[108,5] (modifier) RedundantModifier: Redundant 'strictfp' modifier.
[ERROR] src/test/java/net/sourceforge/pmd/lang/java/ast/ConstantExpressionsTests.java:[207,5] (modifier) RedundantModifier: Redundant 'strictfp' modifier.
[ERROR] src/test/java/net/sourceforge/pmd/lang/java/ast/ConstantExpressionsTests.java:[247,5] (modifier) RedundantModifier: Redundant 'strictfp' modifier.
[ERROR] src/test/java/net/sourceforge/pmd/lang/java/ast/ConstantExpressionsTests.java:[287,5] (modifier) RedundantModifier: Redundant 'strictfp' modifier.
[ERROR] src/test/java/net/sourceforge/pmd/lang/java/ast/ConstantExpressionsTests.java:[302,5] (modifier) RedundantModifier: Redundant 'strictfp' modifier.
[ERROR] src/test/java/net/sourceforge/pmd/lang/java/ast/ConstantExpressionsTests.java:[319,5] (modifier) RedundantModifier: Redundant 'strictfp' modifier.
```

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -137,11 +137,10 @@ workflows:
           name: "no-error-xwiki"
           image-name: "cimg/openjdk:18.0.1"
           command: "./.ci/validation.sh no-error-xwiki"
-      #      until https://github.com/checkstyle/checkstyle/issues/15475
-      #      - validate-with-maven-script:
-      #          name: "no-error-pmd"
-      #          image-name: *cs_img
-      #          command: "./.ci/validation.sh no-error-pmd"
+      - validate-with-maven-script:
+          name: "no-error-pmd"
+          image-name: *cs_img
+          command: "./.ci/validation.sh no-error-pmd"
       - validate-with-maven-script:
           name: "no-exception-struts"
           image-name: *cs_img

